doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -384,6 +422,54 @@ guess the missing piece of information.
 Let-in definitions
 ------------------
 
+.. insertgram term_let exclam_opt
+
+.. productionlist:: coq
+   term_let : let `name` `binder_list_opt` `colon_term_opt` := `term` in `term`
+            : let `single_fix` in `term`
+            : let `names_tuple` `return_type_opt` := `term` in `term`
+            : let ' `pattern` := `term` in `term`
+            : let ' `pattern` := `term` return `term100` in `term`
+            : let ' `pattern` in `pattern` := `term` return `term100` in `term`
+   colon_term_opt : : `term`
+                  : `empty`
+   names_tuple : ( `name_list_comma` )
+               : ()
+   name_list_comma : `name_list_comma` , `name`
+                   : `name`
+   open_binders : `name` `name_list_opt` : `term`
+                : `name` `name_list_opt` `binder_list_opt`
+                : `name` .. `name`
+                : `closed_binder` `binder_list_opt`
+   name_list_opt : `name_list_opt` `name`
+                 : `empty`
+   binder_list_opt : `binder_list_opt` `binder`
+                   : `empty`
+   binder : `name`
+          : `closed_binder`
+   closed_binder : ( `name` `name_list` : `term` )
+                 : ( `name` : `term` )
+                 : ( `name` := `term` )
+                 : ( `name` : `term` := `term` )
+                 : { `name` }
+                 : { `name` `name_list` : `term` }
+                 : { `name` : `term` }
+                 : { `name` `name_list` }
+                 : `( `typeclass_constraint_list_comma` )
+                 : `{ `typeclass_constraint_list_comma` }
+                 : ' `pattern0`
+                 : ( _ : `term` | `term` )
+   name_list : `name_list` `name`
+             : `name`
+   typeclass_constraint_list_comma : `typeclass_constraint_list_comma` , `typeclass_constraint`
+                                   : `typeclass_constraint`
+   typeclass_constraint : ! `term`
+                        : { `name` } : `exclam_opt` `term`
+                        : `name` : `exclam_opt` `term`
+                        : `term`
+   exclam_opt : !
+              : `empty`
```

Comment:
Sorry for the back and forth but the idea was to move only the part on let-in definitions (above). This highlighted chunk should have stayed in the binders section.